### ISSUE TITLE
ArrayIndexOutOfBoundsException when using a Ring of dueling

### ISSUE THREAD

**Issue by emerald000** [Adam-/runelite-plugins#5]
If I use an equipped ring of dueling to teleport, I get the following error. I don't have any pouches in the inventory.

[Client] WARN  n.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: -1
	at hn.getChild(hn.java:18297)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at ew.hr(ew.java:1345)
	at client.aw(client.java:7427)
	at client.aa(client.java:3271)
	at client.v(client.java:1254)
	at bf.ax(bf.java:347)
	at bf.run(bf.java:326)
	at java.lang.Thread.run(Thread.java:748)

**Comment by hjdarnel** [comment]
This also happens when clicking from Fixed to Resizable and back. 
```2020-02-18 00:06:54 [Client] WARN  n.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 10
	at hn.getChild(hn.java:43655)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at ew.hr(ew.java:43351)
	at client.gp(client.java:7498)
	at client.jk(client.java:3271)
	at client.v(client.java:1254)
	at bf.ax(bf.java:347)
	at bf.run(bf.java:326)
	at java.base/java.lang.Thread.run(Thread.java:834)```

**Comment by ThePharros** [comment]
this happens when clicking almost anything (Prayers, equipment, option buttons)
`[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 2
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 9
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 9
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 3
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)
[Client] WARN net.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 10
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)`

**Comment by Hydrox6** [comment]
Went into Kraken Cove, had a crash from this 

```2020-04-22 20:11:56 [Client] WARN  n.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 9
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Unknown Source)
2020-04-22 20:12:43 [Client] WARN  n.runelite.client.eventbus.EventBus - Uncaught exception in event subscriber
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 39
	at he.getChild(he.java:47455)
	at info.sigterm.plugins.esspouch.EssPouchPlugin.onMenuOptionClicked(EssPouchPlugin.java:279)
	at net.runelite.client.eventbus.EventBus$Subscriber.invoke(EventBus.java:73)
	at net.runelite.client.eventbus.EventBus.post(EventBus.java:222)
	at net.runelite.client.callback.Hooks.post(Hooks.java:167)
	at hi.ij(hi.java:11604)
	at client.gc(client.java:8313)
	at client.yj(client.java:3142)
	at client.u(client.java:1113)
	at br.h(br.java:336)
	at br.run(br.java:315)
	at java.base/java.lang.Thread.run(Unknown Source)```

**Comment by Adam-** [comment]
fixed in 91fbe13129d6eb90fe76e118eca176c249f6a285
